### PR TITLE
Con 3346 choose organisation

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/AgreementExtension.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Linq;
+using SFA.DAS.EAS.Account.Api.Types;
+using SFA.DAS.EmployerCommitmentsV2.Web.Models.Cohort;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Extensions
+{
+    public static class AgreementExtension
+    {   
+        public static bool HasSignedAgreement(this LegalEntity legalEntity, bool isTransfer)
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/AgreementExtension.cs:9]: could this name be a bit more obvious as to what it is doing? `HasSignedMinimumRequiredAgreementVersion` or something like that

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Services/EmployerAccountsService.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        }        
+
+        public async Task<List<LegalEntity>> GetLegalEntitiesForAccount(string accountId)
+        {
+            var listOfEntities = await _accountsApiClient.GetLegalEntitiesConnectedToAccount(accountId);
+
+            var list = new List<LegalEntity>();
+            if (listOfEntities.Count == 0) return list;
+
+            list.AddRange(listOfEntities.Select(async entity => await _accountsApiClient.GetLegalEntity(accountId, Convert.ToInt64(entity.Id))).Select(legalEntityViewModel => new LegalEntity
```
Comment (cofaulco) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Services/EmployerAccountsService.cs:27]: could make this parallel to improve the performance although would require some though as some accounts have >900 legal entities. Might be worth raising as a tech debt ticket.

**Review by cofaulco — COMMENTED** [review]
a couple of minor points

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/AgreementExtension.cs (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Extensions/AgreementExtension.cs]: done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Services/EmployerAccountsService.cs (hunk 2 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Services/EmployerAccountsService.cs:27]: tech debt raised : https://skillsfundingagency.atlassian.net/browse/CON-3606

**Review by cofaulco — APPROVED** [review]

**Pull request merged**